src/App.tsx

```diff
@@ -7,33 +8,56 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+type State = {
+  clockName: string,
+  hasClock: boolean,
+  timerId: number,
+};
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    clockName: 'Clock-0',
+    hasClock: true,
+    timerId: 0,
+  };
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.handleClockVisibleFalse);
+    document.addEventListener('click', this.handleClockVisibleTrue);
+
+    this.setState({
+      timerId: window.setInterval(() => {
+        this.setState({ clockName: getRandomName() });
+      }, 3300),
+    });
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.handleClockVisibleFalse);
+    document.removeEventListener('click', this.handleClockVisibleTrue);
+
+    window.clearInterval(this.state.timerId);
+  }
+
+  handleClockVisibleFalse = (event: MouseEvent) => {
+    event.preventDefault();
+    this.setState({ hasClock: false });
+  };
+
+  handleClockVisibleTrue = (event: MouseEvent) => {
+    event.preventDefault();
+    this.setState({ hasClock: true });
+  };
+
```

Comment:
Why do you clear events on your click button? Does this button have special events that need to be cleared?